src/atomics.rs

```diff
@@ -0,0 +1,215 @@
+use core::sync::atomic::*;
+
+pub trait Atomic: Sized {
+    type NonAtomicType: Copy;
```

Comment:
I came across this pull request while I am trying to solve a similar problem. It would be better the `NonAtomicType` is a `PrimInt` (or whatever num trait that is more suitable). In that case, for `NonAtomicType` in the arguments and return values, you can use num traits methods on them.

Reply:
Yeah, that's probably a sensible change